src/server/p2p/network.rs

```diff
@@ -1075,35 +1066,16 @@ where S: ShareChain
                     return;
                 }
 
-                let our_tip_sha3x = self.share_chain_sha3x.chain_pow().await;
-
-                if self.config.sha3x_enabled && response.info.current_sha3x_pow > our_tip_sha3x.as_u128() {
-                    let perform_catch_up_sync = PerformCatchUpSync {
-                        algo: PowAlgorithm::Sha3x,
-                        peer: peer_id,
-                        last_block_from_them: None,
-                        their_height: response.info.current_sha3x_height,
-                        // their_pow: response.info.current_sha3x_pow,
-                        permit: None,
-                    };
-                    let _unused = self
-                        .inner_request_tx
-                        .send(InnerRequest::PerformCatchUpSync(perform_catch_up_sync));
-                }
-                let our_tip_rx = self.share_chain_random_x.chain_pow().await;
-
-                if self.config.randomx_enabled && response.info.current_random_x_pow > our_tip_rx.as_u128() {
-                    let perform_catch_up_sync = PerformCatchUpSync {
-                        algo: PowAlgorithm::RandomX,
-                        peer: peer_id,
-                        last_block_from_them: None,
-                        their_height: response.info.current_random_x_height,
-                        // their_pow: response.info.current_random_x_pow,
-                        permit: None,
-                    };
-                    let _unused = self
-                        .inner_request_tx
-                        .send(InnerRequest::PerformCatchUpSync(perform_catch_up_sync));
```

Comment:
this removal correct?

Reply:
yes, this code is duplicated in the metadata exchange, whereas direct_peer_exchange is only for transferring peers